src/kubernetes_cluster/spec/cluster.rs

```diff
@@ -150,31 +154,66 @@ pub open spec fn kubernetes_api_next<K: ResourceView, R: Reconciler<K>>() -> Act
     }
 }
 
-pub open spec fn controller_next<K: ResourceView, R: Reconciler<K>>() -> Action<State<K, R>, (Option<Message>, Option<ObjectRef>), ()> {
-    let result = |input: (Option<Message>, Option<ObjectRef>), s: State<K, R>| {
-        let host_result = controller::<K, R>().next_result(
-            ControllerActionInput{recv: input.0, scheduled_cr_key: input.1, rest_id_allocator: s.rest_id_allocator},
+pub open spec fn external_api_next<K: ResourceView, E: ExternalAPI, R: Reconciler<K, E>>() -> Action<State<K, E, R>, ExternalComm<E::Input, E::Output>, ()> {
```

Comment:
Could you document this action briefly? Like what I did for `schedule_controller_reconcile` (no need to be that detailed)

Reply:
I don't know how to describe the relations with shim_layer.

Reply:
No, nothing to do with the shim layer. Just briefly describe what this action models in real-world execution.